### DISCUSSION TITLE
X-Axis scale larger than the data points

### DISCUSSION THREAD

**Discussion by alpriest** [patrykandpatrick/vico#459] [Questions] [answered]
My app displays data across the X axis representing values throughout a day. The API returns data partially populated though, so at 9am only midnight -> 9am values are in the dataset. I'd like the X Axis however to be a consistent width always showing 24 hrs. I can't see any way to adjust the scale, and the only way I could conceive would be to enter 0 values to force the graph, but that feels wrong.

Here's the look i'm aiming for.

<img width="449" alt="Screenshot 2023-11-18 at 08 30 01" src="https://github.com/patrykandpatrick/vico/assets/221241/899c33cb-6018-4f04-a0e4-dbf049180d2b">

**Comment by patrickmichalik** [comment, marked as answer]
Hello! You can do this via `AxisValuesOverrider`. Please see the last paragraph of the introduction [here](https://patrykandpatrick.com/vico/wiki/charts). If, for example, you’d like the _x_-axis range to be 0 to 23, you can use `AxisValuesOverrider.fixed(minX = 0f, maxX = 23f)`.

**Comment by alpriest** [comment]
Thanks, completely missed that in the documentation.

**Reply by patrickmichalik** [reply]
You’re welcome!
